File: problem-solving/baekjoon/1764_듣보잡.py
# https://www.acmicpc.net/problem/1764
import sys

if __name__ == "__main__":
    n, m = list(map(int, sys.stdin.readline().split()))
    dbj = {}
    for i in range(n):
        input = sys.stdin.readline().rstrip()
        if dbj.get(input):
            dbj[input] += 1
        else:
            dbj[input] = 1

    for i in range(m):
        input = sys.stdin.readline().rstrip()
        if dbj.get(input):
            dbj[input] += 1
        else:
            dbj[input] = 1

    dbj_result = []
    for (key, value) in dbj.items():
        if value == 2:
            dbj_result.append(key)

    dbj_result.sort()
    print(len(dbj_result))
    for i in dbj_result:
        print(i)

File: problem-solving/baekjoon/1764_듣보잡.go
// https://www.acmicpc.net/problem/1764
package main

import (
	"bufio"
	"fmt"
	"os"
	"sort"
)

func main() {
	reader := bufio.NewReader(os.Stdin)
	writer := bufio.NewWriter(os.Stdout)
	defer writer.Flush()

	var n, m int
	fmt.Fscanln(reader, &n, &m)

	var dbj = map[string]int{}
	for i := 0; i < n; i++ {
		var input string
		fmt.Fscanln(reader, &input)
		dbj[input]++
	}

	for i := 0; i < m; i++ {
		var input string
		fmt.Fscanln(reader, &input)
		dbj[input]++
	}

	keys := []string{}
	for key, val := range dbj {
		if val == 2 {
			keys = append(keys, key)
		}
	}

	sort.Strings(keys)

	fmt.Fprintln(writer, len(keys))
	for _, key := range keys {
		fmt.Fprintln(writer, key)
	}
}
